app/models/manageiq/providers/vmware/network_manager/refresh_parser.rb

```diff
@@ -28,36 +28,70 @@ def ems_inv_to_hashes
 
     private
 
-    def get_networks
-      # fetch org VDC networks
-      @inv[:networks] = @org.networks
+    def get_vdc_networks
+      @inv[:vdc_networks] = @org.networks
+      @inv[:vdc_networks_idx] = Hash[@inv[:vdc_networks].map { |net| [net.id, net] }]
```

Comment:
You can just use `#index_by` here, `@inv[:vdc_networks_idx] = @inv[:vdc_networks].index_by(&:id)`